timemachine/ff/handlers/utils.py

```diff
@@ -67,3 +91,99 @@ def match_smirks(mol, smirks):
         matches.append(tuple(mas))
 
     return matches
+
+
+def check_bond_smarts_symmetric(bond_smarts: str) -> bool:
+    """Match [<atom1>:1]*[<atom2>:2]
+    and return whether atom1 and atom2 are identical strings
+
+    Notes
+    -----
+    * The AM1CCC model contains symmetric patterns that must be assigned 0 parameters
+        (Otherwise, undefined behavior when symmetric bond matches in an arbitrary direction)
+    * Only checks string equivalence!
+        for example
+        check_bond_smarts_symmetric("[#6,#7:1]~[#7,#6:2]")
+        will be a false negative
+    * Does not handle all possible bond smarts
+        for example
+        "[#6:1]~[#6:2]~[#1]"
+        or
+        "[#6:1](~[#8])(~[#16:2])"
+        will not be matched, will return False by default.
+        However, for the bond smarts subset used by the AM1CCC model, this covers most cases
+    """
+
+    pattern = re.compile(r"\[(?P<atom1>.*)\:1\].\[(?P<atom2>.*)\:2\]")
+    match = pattern.match(bond_smarts)
+
+    if type(match) is re.Match:
+        complete = match.span() == (0, len(bond_smarts))
+        symmetric = match.group("atom1") == match.group("atom2")
```

Comment:
Would it be advantageous to do the following?

```suggestion
        symmetric = list(sorted(match.group("atom1").split(","))) == list(sorted(match.group("atom2").split(",")))
```

Seems to work in the case of `[#6,#7:1]~[#7,#6:2]`

That said, nothing in our AM1BCC charges seems to have this lack of ordering, so probably unimportant

Reply:
Can double-check this -- one issue with `.split(",")` is that comma-separated lists can appear inside of nested structures. For example, applying `match.group("atom1").split(",")`  to this pattern https://github.com/proteneer/timemachine/blob/fd14908113315ca07c8983e7ecd4dd92178d03a8/timemachine/ff/params/smirnoff_1_1_0_ccc.py#L346

returns `['#6X3$(*=[#7', '#15])']` which looks undesirable -- would probably want to sort recursively

However, there are other ways you could produce syntactically distinct SMARTS with identical meaning. Not sure if there's a robust "canonicalize SMARTS" function...

Reply:
Increasingly disliking the regex approach